### PR TITLE
[SPARK-14518] [SQL] Support Comment in CREATE VIEW

### PR DESCRIPTION
**Pull request by gatorsmile**
#### What changes were proposed in this pull request?

**HQL Syntax**: [Create View](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-Create/Drop/AlterView)

``` SQL
CREATE VIEW [IF NOT EXISTS] [db_name.]view_name [(column_name [COMMENT column_comment], ...) ]
  [COMMENT view_comment]
  [TBLPROPERTIES (property_name = property_value, ...)]
  AS SELECT ...;
```

Add a support for the `[COMMENT view_comment]` clause
#### How was this patch tested?

Modified the existing test cases to verify the correctness.

### CODE REVIEW

**Comment by gatorsmile** [comment]
cc @yhuai @andrewor14 Thanks!

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveSqlParser.scala [hunk 1]
```diff
@@ -257,18 +252,20 @@ class HiveSqlAstBuilder extends SparkSqlAstBuilder {
   private def createView(
       ctx: ParserRuleContext,
       name: TableIdentifierContext,
+      description: Option[String],
```
Comment (yhuai) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveSqlParser.scala]: Should we just call it comment? Or, are we using description in other places?

Comment (gatorsmile) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveSqlParser.scala]: https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala#L268

`CatalogDatabase` uses `description` as `comment`. Maybe comment is more straightforward. Let me change it. Thanks!

**Comment by yhuai** [comment]
@gatorsmile Thank you for working on it. How about we hold off the change until we merge https://github.com/apache/spark/pull/12271/files? That PR is adding the field for comment in the CatalogTable.

**Comment by gatorsmile** [comment]
Sure, no problem. : )

**Comment by gatorsmile** [comment]
@yhuai @andrewor14 @cloud-fan @liancheng 

After the merge of `Create Table`, this PR is changed by using the latest `CatalogTable `. It is ready for review. Thanks!

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala [hunk 2]
```diff
@@ -110,6 +110,24 @@ class HiveDDLSuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
...
+    val catalog = hiveContext.sessionState.catalog
+    val tabName = "tab1"
+    withTable(tabName) {
+      sql(s"CREATE TABLE $tabName(c1 int) COMMENT 'BLABLA'")
+      val viewName = "view1"
+      withView(viewName) {
+        sql(s"CREATE VIEW $viewName COMMENT 'no comment' AS SELECT * FROM $tabName")
+        val tableMetadata = catalog.getTableMetadata(TableIdentifier(tabName, Some("default")))
+        val viewMetadata = catalog.getTableMetadata(TableIdentifier(viewName, Some("default")))
+        assert(tableMetadata.properties.filter(_._1 != "transient_lastDdlTime")
```
Comment (cloud-fan) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala]: I'd like to do `assert(tableMetadata.properties.get("comment") == "BLABLA")`

**Comment by cloud-fan** [comment]
LGTM except one minor comment

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala (hunk 2 above)
Comment (gatorsmile) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala]: Sure, do it now. Thanks!

**Comment by yhuai** [comment]
Thanks. Merging to master.

**Pull request closed without merging**